test/e2e/framework/util.go

```diff
@@ -1165,12 +1165,14 @@ func hasRemainingContent(c clientset.Interface, clientPool dynamic.ClientPool, n
 	}
 
 	// find out what content is supported on the server
-	resources, err := c.Discovery().ServerPreferredNamespacedResources()
-	if err != nil && !isDynamicDiscoveryError(err) {
+	// Since extension apiserver is not always available, e.g. metrics server sometimes goes down,
+	// add retry here.
+	resources, err := WaitForServerPreferredNamespacedResources(c.Discovery(), 30*time.Second)
+	if err != nil {
 		return false, err
 	}
 	groupVersionResources, err := discovery.GroupVersionResources(resources)
```

Comment:
Will the metrics server go down between line 1170 and here?

Reply:
GroupVersionResources is only a helper function. It doesn't try to access any resource. :)

Reply:
To @caesarxuchao's point, I wonder if you would be better of doing this with a wait.Poll instead? Eg. 
    var groupVersionResources
    err = wait.Poll(100*time.Millisecond, 30*time.Second, func() (bool, error) {
		groupVersionResources, err = discovery.GroupVersionResources(resources)
 	    if err != nil {
  		    return false, err
  	    }
        return true, nil
    })
    if err != nil {
	     return false, err
    }

Reply:
Since this is a helper function, there is no server access, just some type conversion. I prefer return error directly instead of retrying it. :)

Reply:
> Will the metrics server go down between line 1170 and here?

I agree... I'm not sure a point in time check gets you a whole lot

Reply:
@liggitt The following code only generates log instead of returning error. (Other than this line https://github.com/kubernetes/kubernetes/pull/55764#discussion_r151282698)